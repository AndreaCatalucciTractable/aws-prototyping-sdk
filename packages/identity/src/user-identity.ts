import {
  IdentityPool,
  IdentityPoolProps,
  UserPoolAuthenticationProvider,
} from "@aws-cdk/aws-cognito-identitypool-alpha";
import { UserPool, UserPoolClient } from "aws-cdk-lib/aws-cognito";
import { Construct } from "constructs";

/**
 * Properties which configures the Identity Pool.
 */
export interface UserIdentityProps {
  /**
   * User provided Cognito UserPool.
   *
   * @default - a userpool will be created.
   */
  readonly userPool?: UserPool;

  /**
   * Configuration for the Identity Pool.
   */
  readonly identityPoolOptions?: IdentityPoolProps;
}

/**
 * Creates an Identity Pool with sane defaults configured.
 */
export class UserIdentity extends Construct {
  public readonly userPool?: UserPool;
  public readonly userPoolClient?: UserPoolClient;
  public readonly identityPool: IdentityPool;

  constructor(scope: Construct, id: string, props?: UserIdentityProps) {
    super(scope, id);

    // Unless explicitly stated, created a default Cognito User Pool and Web Client.
    if (!props?.userPool) {
      this.userPool = new UserPool(this, "UserPool");
      this.userPoolClient = this.userPool.addClient("WebClient", {
        authFlows: {
          userPassword: true,
          userSrp: true,
        },
      });
    }

    this.identityPool = new IdentityPool(this, "IdentityPool", {
      ...props?.identityPoolOptions,
      authenticationProviders: {
        ...props?.identityPoolOptions?.authenticationProviders,
        userPools: [
          ...(props?.identityPoolOptions?.authenticationProviders?.userPools ||
            []),
          ...(this.userPool
            ? [new UserPoolAuthenticationProvider({ userPool: this.userPool })]
            : []),
        ],
      },
    });
  }
}
