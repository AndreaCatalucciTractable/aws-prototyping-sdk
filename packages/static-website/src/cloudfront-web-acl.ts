import { Names, Stack } from "aws-cdk-lib";
import {
  AwsCustomResource,
  AwsCustomResourcePolicy,
  PhysicalResourceId,
  PhysicalResourceIdReference,
} from "aws-cdk-lib/custom-resources";
import { WAFV2 } from "aws-sdk";
import { Construct } from "constructs";

const ACL_REGION = "us-east-1";
const SCOPE = "CLOUDFRONT";

/**
 * Represents a WAF V2 managed rule.
 */
export interface ManagedRule {
  /**
   * The name of the managed rule group vendor. You use this, along with the rule group name, to identify the rule group.
   */
  readonly vendor: string;

  /**
   * The name of the managed rule group. You use this, along with the vendor name, to identify the rule group.
   */
  readonly name: string;
}

/**
 * Properties to configure the web acl.
 */
export interface CloudFrontWebAclProps {
  /**
   * List of managed rules to apply to the web acl.
   */
  readonly managedRules: ManagedRule[];
}

/**
 * This construct creates a WAFv2 Web ACL for cloudfront in the us-east-1 region (required for cloudfront) no matter the
 * region of the parent cdk stack.
 */
export class CloudfrontWebAcl extends Construct {
  public readonly webAclId: string;
  public readonly webAclArn: string;
  public readonly name: string;
  public readonly region: string = ACL_REGION;

  constructor(scope: Construct, id: string, props: CloudFrontWebAclProps) {
    super(scope, id);

    this.name = `${id.substring(0, 40)}_${Names.uniqueId(this)}`;

    // The parameters for creating the Web ACL
    const createWebACLRequest: WAFV2.Types.CreateWebACLRequest = {
      Name: this.name,
      DefaultAction: { Allow: {} },
      Scope: SCOPE,
      VisibilityConfig: {
        CloudWatchMetricsEnabled: true,
        MetricName: id,
        SampledRequestsEnabled: true,
      },
      Rules: props.managedRules
        .map((r) => ({ VendorName: r.vendor, Name: r.name }))
        .map((rule, Priority) => ({
          Name: `${rule.VendorName}-${rule.Name}`,
          Priority,
          Statement: { ManagedRuleGroupStatement: rule },
          OverrideAction: { None: {} },
          VisibilityConfig: {
            MetricName: `${rule.VendorName}-${rule.Name}`,
            CloudWatchMetricsEnabled: true,
            SampledRequestsEnabled: true,
          },
        })),
    };

    // Create the Web ACL
    const createCustomResource = new AwsCustomResource(this, `Create`, {
      policy: AwsCustomResourcePolicy.fromSdkCalls({
        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
      }),
      onCreate: {
        service: "WAFV2",
        action: "createWebACL",
        parameters: createWebACLRequest,
        region: this.region,
        physicalResourceId: PhysicalResourceId.fromResponse("Summary.Id"),
      },
    });
    this.webAclId = createCustomResource.getResponseField("Summary.Id");

    const getWebACLRequest: WAFV2.Types.GetWebACLRequest = {
      Name: this.name,
      Scope: SCOPE,
      Id: this.webAclId,
    };

    // A second custom resource is used for managing the deletion of this construct, since both an Id and LockToken
    // are required for Web ACL Deletion
    new AwsCustomResource(this, `Delete`, {
      policy: AwsCustomResourcePolicy.fromSdkCalls({
        resources: AwsCustomResourcePolicy.ANY_RESOURCE,
      }),
      onCreate: {
        service: "WAFV2",
        action: "getWebACL",
        parameters: getWebACLRequest,
        region: this.region,
        physicalResourceId: PhysicalResourceId.fromResponse("LockToken"),
      },
      onDelete: {
        service: "WAFV2",
        action: "deleteWebACL",
        parameters: {
          Name: this.name,
          Scope: SCOPE,
          Id: this.webAclId,
          LockToken: new PhysicalResourceIdReference(),
        },
        region: this.region,
      },
    });
    this.webAclArn = `arn:aws:wafv2:${this.region}:${
      Stack.of(this).account
    }:global/webacl/${this.name}/${this.webAclId}`;
  }
}
