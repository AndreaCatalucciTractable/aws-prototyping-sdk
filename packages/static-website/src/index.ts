export * from "./cloudfront-web-acl";
export * from "./static-website-auth";
export * from "./static-website";
