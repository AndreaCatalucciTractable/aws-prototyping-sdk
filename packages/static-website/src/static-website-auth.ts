import {
  IdentityPool,
  IdentityPoolProps,
  UserPoolAuthenticationProvider,
} from "@aws-cdk/aws-cognito-identitypool-alpha";
import {
  UserPool,
  UserPoolClient,
  UserPoolClientOptions,
  UserPoolProps,
} from "aws-cdk-lib/aws-cognito";
import { Construct } from "constructs";

const DEFAULT_COGNITO_USER_POOL_CLIENT_OPTIONS: UserPoolClientOptions = {
  authFlows: {
    userPassword: true,
    userSrp: true,
  },
};

/**
 * Default Cognito User Pool options.
 */
export interface CognitoUserPoolOptions {
  /**
   * Props for UserPool.
   */
  readonly userPoolOptions?: UserPoolProps;

  /**
   * Props for UserPoolClient.
   */
  readonly userPoolClientOptions?: UserPoolClientOptions;
}

/**
 * Properties which configures the Identity Pool and optional Cognito User Pool.
 */
export interface StaticWebsiteAuthProps {
  /**
   * Determines whether to create a default Cognito User Pool.
   *
   * @default - true
   */
  readonly createCognitoUserPool?: boolean;

  /**
   * Configuration for the default User Pool.
   *
   * Only used when createCognitoUserPool is not false.
   */
  readonly cognitoUserPoolOptions?: CognitoUserPoolOptions;

  /**
   * Configuration for the Identity Pool.
   */
  readonly identityPoolOptions?: IdentityPoolProps;
}

/**
 * Creates an Identity Pool with sane defaults configured.
 */
export class StaticWebsiteAuth extends Construct {
  public readonly userPool?: UserPool;
  public readonly userPoolClient?: UserPoolClient;
  public readonly identityPool: IdentityPool;

  constructor(scope: Construct, id: string, props?: StaticWebsiteAuthProps) {
    super(scope, id);

    // Unless explicitly stated, created a default Cognito User Pool and Web Client.
    if (props?.createCognitoUserPool !== false) {
      this.userPool = new UserPool(
        this,
        "UserPool",
        props?.cognitoUserPoolOptions?.userPoolOptions
      );
      this.userPoolClient = this.userPool.addClient(
        "WebClient",
        props?.cognitoUserPoolOptions?.userPoolClientOptions ||
          DEFAULT_COGNITO_USER_POOL_CLIENT_OPTIONS
      );
    }

    this.identityPool = new IdentityPool(this, "IdentityPool", {
      ...props?.identityPoolOptions,
      authenticationProviders: {
        ...props?.identityPoolOptions?.authenticationProviders,
        userPools: [
          ...(props?.identityPoolOptions?.authenticationProviders?.userPools ||
            []),
          ...(this.userPool
            ? [new UserPoolAuthenticationProvider({ userPool: this.userPool })]
            : []),
        ],
      },
    });
  }
}
