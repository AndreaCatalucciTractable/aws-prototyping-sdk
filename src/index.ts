// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

export * as pdk_pipeline from "./pdk_pipeline";
export * as pdk_projen from "./pdk_projen";
