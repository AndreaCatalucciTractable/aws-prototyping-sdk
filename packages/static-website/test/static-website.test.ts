import { App, Stack } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { StaticWebsite } from "../src";

describe("Static Website Unit Tests", () => {
  it("Defaults", () => {
    const app = new App();
    const stack = new Stack(app);
    new StaticWebsite(stack, "Defaults", {
      websiteContentPath: ".",
    });

    expect(Template.fromStack(stack)).toMatchSnapshot();
  });
});
