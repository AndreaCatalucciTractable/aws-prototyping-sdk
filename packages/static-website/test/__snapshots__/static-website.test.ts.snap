// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static Website Unit Tests Defaults 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "AWS679f53fac002430cb0da5b7982bd22872D164C4C": Object {
      "DependsOn": Array [
        "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "9ab58259097ac1249b4061ac6e823f78810b11c4c760a30324f6d012fe8e6440.zip",
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 120,
      },
      "Type": "AWS::Lambda::Function",
    },
    "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": Object {
      "DependsOn": Array [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "f98b78092dcdd31f5e6d47489beb5f804d4835ef86a8085d0a2053cb9ae711da.zip",
        },
        "Handler": "index.handler",
        "Layers": Array [
          Object {
            "Ref": "DefaultsWebsiteDeploymentAwsCliLayerD5AA12CC",
          },
        ],
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.7",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DefaultsWebsiteBucket3263D025",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DefaultsWebsiteBucket3263D025",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "DefaultsWebsiteBucketKeyAD72391B",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "cloudfront:GetInvalidation",
                "cloudfront:CreateInvalidation",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": Array [
          Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": Object {
      "DependsOn": Array [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "0758db37a61ce7177f50559cad0340df233a2001b5880615afa517203dd67c21.zip",
        },
        "Description": Object {
          "Fn::Join": Array [
            "",
            Array [
              "Lambda function for auto-deleting objects in ",
              Object {
                "Ref": "DefaultsWebsiteBucket3263D025",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "DefaultsCloudfrontDistributionF4EA1054": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "CustomErrorResponses": Array [
            Object {
              "ErrorCode": 404,
              "ResponseCode": 200,
              "ResponsePagePath": "/index.html",
            },
          ],
          "DefaultCacheBehavior": Object {
            "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
            "Compress": true,
            "TargetOriginId": "DefaultsCloudfrontDistributionOrigin1BA23CD94",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "DefaultsWebsiteBucket3263D025",
                  "RegionalDomainName",
                ],
              },
              "Id": "DefaultsCloudfrontDistributionOrigin1BA23CD94",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "origin-access-identity/cloudfront/",
                      Object {
                        "Ref": "DefaultsCloudfrontDistributionOrigin1S3OriginCFA7596E",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "WebACLId": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:aws:wafv2:us-east-1:",
                Object {
                  "Ref": "AWS::AccountId",
                },
                ":global/webacl/WebsiteAcl_DefaultsWebsiteAcl851D3A03/",
                Object {
                  "Fn::GetAtt": Array [
                    "DefaultsWebsiteAclCreateD833F225",
                    "Summary.Id",
                  ],
                },
              ],
            ],
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "DefaultsCloudfrontDistributionOrigin1S3OriginCFA7596E": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": "Identity for DefaultsCloudfrontDistributionOrigin1BA23CD94",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "DefaultsWebsiteAclCreateCustomResourcePolicy7011FDE8": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "wafv2:CreateWebACL",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DefaultsWebsiteAclCreateCustomResourcePolicy7011FDE8",
        "Roles": Array [
          Object {
            "Ref": "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DefaultsWebsiteAclCreateD833F225": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "DefaultsWebsiteAclCreateCustomResourcePolicy7011FDE8",
      ],
      "Properties": Object {
        "Create": "{\\"service\\":\\"WAFV2\\",\\"action\\":\\"createWebACL\\",\\"parameters\\":{\\"Name\\":\\"WebsiteAcl_DefaultsWebsiteAcl851D3A03\\",\\"DefaultAction\\":{\\"Allow\\":{}},\\"Scope\\":\\"CLOUDFRONT\\",\\"VisibilityConfig\\":{\\"CloudWatchMetricsEnabled\\":true,\\"MetricName\\":\\"WebsiteAcl\\",\\"SampledRequestsEnabled\\":true},\\"Rules\\":[{\\"Name\\":\\"AWS-AWSManagedRulesCommonRuleSet\\",\\"Priority\\":0,\\"Statement\\":{\\"ManagedRuleGroupStatement\\":{\\"VendorName\\":\\"AWS\\",\\"Name\\":\\"AWSManagedRulesCommonRuleSet\\"}},\\"OverrideAction\\":{\\"None\\":{}},\\"VisibilityConfig\\":{\\"MetricName\\":\\"AWS-AWSManagedRulesCommonRuleSet\\",\\"CloudWatchMetricsEnabled\\":true,\\"SampledRequestsEnabled\\":true}}]},\\"region\\":\\"us-east-1\\",\\"physicalResourceId\\":{\\"responsePath\\":\\"Summary.Id\\"}}",
        "InstallLatestAwsSdk": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "AWS679f53fac002430cb0da5b7982bd22872D164C4C",
            "Arn",
          ],
        },
      },
      "Type": "Custom::AWS",
      "UpdateReplacePolicy": "Delete",
    },
    "DefaultsWebsiteAclDelete7CCC03C8": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "DefaultsWebsiteAclDeleteCustomResourcePolicy9938C27E",
      ],
      "Properties": Object {
        "Create": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{\\"service\\":\\"WAFV2\\",\\"action\\":\\"getWebACL\\",\\"parameters\\":{\\"Name\\":\\"WebsiteAcl_DefaultsWebsiteAcl851D3A03\\",\\"Scope\\":\\"CLOUDFRONT\\",\\"Id\\":\\"",
              Object {
                "Fn::GetAtt": Array [
                  "DefaultsWebsiteAclCreateD833F225",
                  "Summary.Id",
                ],
              },
              "\\"},\\"region\\":\\"us-east-1\\",\\"physicalResourceId\\":{\\"responsePath\\":\\"LockToken\\"}}",
            ],
          ],
        },
        "Delete": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{\\"service\\":\\"WAFV2\\",\\"action\\":\\"deleteWebACL\\",\\"parameters\\":{\\"Name\\":\\"WebsiteAcl_DefaultsWebsiteAcl851D3A03\\",\\"Scope\\":\\"CLOUDFRONT\\",\\"Id\\":\\"",
              Object {
                "Fn::GetAtt": Array [
                  "DefaultsWebsiteAclCreateD833F225",
                  "Summary.Id",
                ],
              },
              "\\",\\"LockToken\\":\\"PHYSICAL:RESOURCEID:\\"},\\"region\\":\\"us-east-1\\"}",
            ],
          ],
        },
        "InstallLatestAwsSdk": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "AWS679f53fac002430cb0da5b7982bd22872D164C4C",
            "Arn",
          ],
        },
      },
      "Type": "Custom::AWS",
      "UpdateReplacePolicy": "Delete",
    },
    "DefaultsWebsiteAclDeleteCustomResourcePolicy9938C27E": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "wafv2:GetWebACL",
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": "wafv2:DeleteWebACL",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DefaultsWebsiteAclDeleteCustomResourcePolicy9938C27E",
        "Roles": Array [
          Object {
            "Ref": "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "DefaultsWebsiteBucket3263D025": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "KMSMasterKeyID": Object {
                  "Fn::GetAtt": Array [
                    "DefaultsWebsiteBucketKeyAD72391B",
                    "Arn",
                  ],
                },
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "LoggingConfiguration": Object {
          "LogFilePrefix": "access-logs",
        },
        "Tags": Array [
          Object {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
          Object {
            "Key": "aws-cdk:cr-owned:fb67e0d5",
            "Value": "true",
          },
        ],
        "VersioningConfiguration": Object {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "DefaultsWebsiteBucketAutoDeleteObjectsCustomResourceD840F8F2": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "DefaultsWebsiteBucketPolicy594E6643",
      ],
      "Properties": Object {
        "BucketName": Object {
          "Ref": "DefaultsWebsiteBucket3263D025",
        },
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "DefaultsWebsiteBucketKeyAD72391B": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "Description": "Created by Default/Defaults/WebsiteBucket",
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": Array [
          Object {
            "Key": "aws-cdk:cr-owned:fb67e0d5",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "DefaultsWebsiteBucketPolicy594E6643": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "DefaultsWebsiteBucket3263D025",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:*",
              "Condition": Object {
                "Bool": Object {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": Object {
                "AWS": "*",
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DefaultsWebsiteBucket3263D025",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DefaultsWebsiteBucket3263D025",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DefaultsWebsiteBucket3263D025",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DefaultsWebsiteBucket3263D025",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": Object {
                "CanonicalUser": Object {
                  "Fn::GetAtt": Array [
                    "DefaultsCloudfrontDistributionOrigin1S3OriginCFA7596E",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "DefaultsWebsiteBucket3263D025",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "DefaultsWebsiteDeploymentAwsCliLayerD5AA12CC": Object {
      "Properties": Object {
        "Content": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "8ad7bbf8be94e05d569da95ddb82511dcc959f25054825394cbb86028ccd1b6a.zip",
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "DefaultsWebsiteDeploymentCustomResource326CD6C1": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DestinationBucketName": Object {
          "Ref": "DefaultsWebsiteBucket3263D025",
        },
        "DistributionId": Object {
          "Ref": "DefaultsCloudfrontDistributionF4EA1054",
        },
        "Prune": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": Array [
          Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
        ],
        "SourceObjectKeys": Array [
          "a79f62b4071246acc1e8e834ba67dc3bbf15a3662e39d31667fa59315ef86f56.zip",
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
